File: app/src/main/java/com/overswayit/plesnisavezsrbije/models/AgeCategory.java
package com.overswayit.plesnisavezsrbije.models;

/**
 * Created by lazarristic on 25/02/2019.
 * Copyright (c) 2019 PlesniSavezSrbije. All rights reserved.
 */
public enum AgeCategory {
    JUVENILE("PIO"),
    JUNIOR_I("MLO"),
    JUNIOR_II("OML"),
    YOUTH("STO"),
    ADULT("SEN"),
    SENIOR("STS");

    private String val;

    AgeCategory(String val) {
        this.val = val;
    }

    public static AgeCategory fromString(String category) {
        for (AgeCategory ageCategory : AgeCategory.values()) {
            if (ageCategory.val.equals(category)) {
                return ageCategory;
            }
        }

        return null;
    }

    public String asString() {
        return this.val;
    }
}

File: app/src/main/java/com/overswayit/plesnisavezsrbije/models/AgeCategory.kt
package com.overswayit.plesnisavezsrbije.models

/**
 * Created by lazarristic on 25/02/2019.
 * Copyright (c) 2019 PlesniSavezSrbije. All rights reserved.
 */
enum class AgeCategory private constructor(private val `val`: String) {
    JUVENILE("PIO"),
    JUNIOR_I("MLO"),
    JUNIOR_II("OML"),
    YOUTH("STO"),
    ADULT("SEN"),
    SENIOR("STS");

    fun asString(): String {
        return this.`val`
    }

    companion object {

        fun fromString(category: String): AgeCategory? {
            for (ageCategory in values()) {
                if (ageCategory.`val` == category) {
                    return ageCategory
                }
            }

            return null
        }
    }
}
